### DISCUSSION TITLE
Messages Held - Disk space

### DISCUSSION THREAD

**Discussion by mattyg2301** [postalserver/postal#2929] [Help with using Postal]
Good day,

Does anyone know how I can flush the messages held queue, I've set the retention to 1 day and have run the below command from postal console but the disk space on the server just keeps growing.

The server is working fine and outbound mails are being sent so not sure how to reset everything again.

Server['ARC/ARC'].message_db.messages(:where => {:held => 1, :scope => 'outgoing'}).each { |m| m.add_to_message_queue(:manual => true) }
![image](https://github.com/postalserver/postal/assets/25392804/2279f753-7ee9-4428-a443-dfdfdceed12b)

Any advice would be greatly appreciated.

**Comment by willpower232** [comment]
I think the messages are only cleared up once per day so if you're adding more held messages before the crons happen then it might be a matter of time to alleviate the pressure unless you need the disk space back urgently in which case you would have to connect to the database directly to see where the space was being used.

**Reply by mattyg2301** [reply]
Hi Will,

Thanks for your response.

These messages have been growing daily, we stopped sending messages to this server to see if it would clear but it doesn't. It just seems to be growing and growing.

**Reply by willpower232** [reply]
that shouldn't be the case if nothing is using it :sweat_smile: 

are you able to connect to the database and see what tables you have?

**Reply by mattyg2301** [reply]
Hi Will, 

See below and let me know if you need anything else.

MariaDB [postal]> SHOW TABLES;
+----------------------------+
| Tables_in_postal           |
+----------------------------+
| additional_route_endpoints |
| address_endpoints          |
| ar_internal_metadata       |
| authie_sessions            |
| credentials                |
| domains                    |
| http_endpoints             |
| ip_addresses               |
| ip_pool_rules              |
| ip_pools                   |
| organization_ip_pools      |
| organization_users         |
| organizations              |
| queued_messages            |
| routes                     |
| scheduled_tasks            |
| schema_migrations          |
| servers                    |
| smtp_endpoints             |
| statistics                 |
| track_certificates         |
| track_domains              |
| user_invites               |
| users                      |
| webhook_events             |
| webhook_requests           |
| webhooks                   |
| worker_roles               |
+----------------------------+

Thanks

**Reply by mattyg2301** [reply]
Not sure if this helps but these are the files using all the space.

--- /var/lib/docker/volumes/cc4b5bc9f15e002beee5d80614476db095262277ab4beb68bc2045245b56c7eb/_data/postal@002dserver@002d1 --------------------------------------------------------------------------------------------------------------------------------------------------                         /..                                                                                                                                                                                                                                                 
    6.7 GiB [##########]  raw@002d2024@002d04@002d04.ibd
    4.9 GiB [#######   ]  raw@002d2024@002d03@002d26.ibd
    4.9 GiB [#######   ]  raw@002d2024@002d03@002d28.ibd
    3.8 GiB [#####     ]  raw@002d2024@002d04@002d05.ibd
    3.3 GiB [####      ]  deliveries.ibd
    3.3 GiB [####      ]  raw@002d2024@002d04@002d03.ibd
    2.5 GiB [###       ]  raw@002d2024@002d03@002d27.ibd
    2.4 GiB [###       ]  raw@002d2024@002d04@002d02.ibd
    2.4 GiB [###       ]  raw@002d2024@002d04@002d16.ibd
    2.4 GiB [###       ]  raw@002d2024@002d04@002d15.ibd
    2.4 GiB [###       ]  raw@002d2024@002d04@002d10.ibd
    2.4 GiB [###       ]  raw@002d2024@002d04@002d18.ibd
    2.4 GiB [###       ]  raw@002d2024@002d04@002d11.ibd
    2.1 GiB [###       ]  raw@002d2024@002d03@002d25.ibd
    1.2 GiB [#         ]  raw@002d2024@002d04@002d08.ibd
    1.0 GiB [#         ]  messages.ibd

**Reply by willpower232** [reply]
You'll need to track down the specific database that represents this server within Postal (you should see several) and see how many messages tables you have.

**Reply by mattyg2301** [reply]
Thanks for getting back to me, I probably should have mentioned this in the beginning but this instance of postal only has one server.

It's a very low volume setup for a single customer

**Reply by willpower232** [reply]
so you're looking for the only other database with `postal` at the start of the name

**Reply by mattyg2301** [reply]
Ah, got you, see below:

MariaDB [postal-server-1]> SHOW TABLES;
+---------------------------+
| Tables_in_postal-server-1 |
+---------------------------+
| clicks                    |
| deliveries                |
| links                     |
| live_stats                |
| loads                     |
| messages                  |
| migrations                |
| raw-2024-03-25            |
| raw-2024-03-26            |
| raw-2024-03-27            |
| raw-2024-03-28            |
| raw-2024-03-29            |
| raw-2024-03-30            |
| raw-2024-03-31            |
| raw-2024-04-01            |
| raw-2024-04-02            |
| raw-2024-04-03            |
| raw-2024-04-04            |
| raw-2024-04-05            |
| raw-2024-04-06            |
| raw-2024-04-07            |
| raw-2024-04-08            |
| raw-2024-04-09            |
| raw-2024-04-10            |
| raw-2024-04-11            |
| raw-2024-04-12            |
| raw-2024-04-13            |
| raw-2024-04-14            |
| raw-2024-04-15            |
| raw-2024-04-16            |
| raw-2024-04-17            |
| raw-2024-04-18            |
| raw-2024-04-19            |
| raw_message_sizes         |
| spam_checks               |
| stats_daily               |
| stats_hourly              |
| stats_monthly             |
| stats_yearly              |
| suppressions              |
| webhook_requests          |
+---------------------------+
41 rows in set (0.001 sec)

**Reply by willpower232** [reply]
did you change both retention settings to 1 day?

**Reply by mattyg2301** [reply]
Yes, both are set to 1 day, been like that for weeks now.

**Reply by willpower232** [reply]
Weird, well if you start by deleting the oldest raw table and see if that prompts it back into working, perhaps it is just getting stuck on some level. Presumably you don't want to keep managing this directly so I'd say try deleting one table a day and see if Postal can sort itself out eventually.

**Reply by mattyg2301** [reply]
Perfect thanks, will give that a shot and fingers crossed it sorts itself out.

Thanks for your assistance!
